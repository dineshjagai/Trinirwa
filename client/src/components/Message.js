/* eslint-disable jsx-a11y/media-has-caption */
import 'bootstrap/dist/css/bootstrap.css';
import React, { useEffect, useState } from 'react';
import Divider from '@material-ui/core/Divider';
import './tweet.css';
import {
  getAvatar,
} from './Module';
import { getCurrentUsername } from '../auth/authServices';

export default function Message({ data }) {
  const user = getCurrentUsername();
  // const receiver = getCurrentReceiver();
  const [avatar, setAvatar] = useState('');
  // const [hadRead, setHasRead] = useState(false);
  // eslint-disable-next-line eqeqeq
  const [isPicture] = useState(data.type === 'picture');
  const [isVideo] = useState(data.type === 'video');
  const [isSong] = useState(data.type === 'song');

  const getData = async () => {
    await getAvatar(data.user).then((res) => {
      setAvatar(res.data.avatar[0].profile_picture);
    }).catch((err) => {
      console.log(err.message);
    });
  };

  useEffect(() => {
    getData();
  }, []);

  const date = (data.message_date.split('T'))[0];
  const newAvatar = `/api/viewFile/${avatar}`;
  const newMedia = `/api/viewFile/${data.content}`;

  let newMediaMessage;
  if (isPicture) {
    newMediaMessage = <img className="tweet_image" id="tweet_media_picture" src={newMedia} alt="" />;
  } else if (isVideo) {
    newMediaMessage = (
      <video className="tweet_video" controls>
        {' '}
        <source src={newMedia} />
      </video>
    );
  } else if (isSong) {
    console.log('song');
    newMediaMessage = (
      <audio controls>
        <source src={newMedia} />
      </audio>
    );
  } else {
    newMediaMessage = data.content;
  }
  // eslint-disable-next-line eqeqeq
  const hasRead = (data.is_read == 1);
  let messageStatus = 'Message Status : Delivered';
  if (hasRead) {
    messageStatus = 'Message Status : Read';
  }

  let senderReceiver = 'Receiver>';
  if (user === data.user) {
    senderReceiver = 'Sender>';
  } else {
    messageStatus = '';
  }
  return (

    <div id="container_tweet">
      <div className="tweet_header">
        <div className="tweet_img">
          <img className="tweet_header" id="tweet_author_picture" src={newAvatar} alt="" />
        </div>
        <div className="tweet_text">
          <span
            style={{
              fontSize: '13px',
              fontWeight: 'bold',
            }}
            id="author_username"
          >
            {senderReceiver}
            {' '}
            {data.user}

          </span>
          <span
            style={{
              fontSize: '13px',
              fontWeight: 'bold',
            }}
            id="author_message"

          >
            {' '}
            {messageStatus}

          </span>
          <span style={{ fontSize: '10px' }} id="date">
            {date}
          </span>
        </div>
      </div>

      <Divider variant="middle" />
      <div className="tweet_content">
        <p style={{ textAlign: 'left' }}>
          {newMediaMessage}
        </p>
      </div>
      <Divider variant="middle" />

    </div>

  );
}
