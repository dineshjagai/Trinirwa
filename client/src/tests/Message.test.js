import React from 'react';
import renderer from 'react-test-renderer';
import Message from '../components/Message';

test('Test Message snapshot', () => {
  const d = {
    'user' : 'isimbib',
    'message_date' :  '2008-09-07',
    'content' : 'Hiii',
    'tweet_likes': 0,
    'tweet_id' :  '1',
    'type' : 'picture',
    'tweet_blocks': 0,
  };
  const component = renderer.create(<Message data={d}/>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
