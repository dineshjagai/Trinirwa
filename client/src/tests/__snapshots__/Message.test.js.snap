// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Message snapshot 1`] = `
<div
  id="container_tweet"
>
  <div
    className="tweet_header"
  >
    <div
      className="tweet_img"
    >
      <img
        alt=""
        className="tweet_header"
        id="tweet_author_picture"
        src="/api/viewFile/"
      />
    </div>
    <div
      className="tweet_text"
    >
      <span
        id="author_username"
        style={
          Object {
            "fontSize": "13px",
            "fontWeight": "bold",
          }
        }
      >
        isimbib
      </span>
      <span
        id="date"
        style={
          Object {
            "fontSize": "10px",
          }
        }
      >
        2008-09-07
      </span>
    </div>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-middle"
  />
  <div
    className="tweet_content"
  >
    <p
      style={
        Object {
          "textAlign": "left",
        }
      }
    >
      <img
        alt=""
        className="tweet_image"
        id="tweet_media_picture"
        src="/api/viewFile/Hiii"
      />
    </p>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-middle"
  />
</div>
`;
